app/controllers/journeys/expenses/workplaceRunningCosts/WorkplaceRunningCostsCYAController.scala

```diff
@@ -0,0 +1,133 @@
+              WfhFlatRateOrActualCostsSummary.row(request.userAnswers, taxYear, businessId, user, wfhRate.getOrElse("")),
+              WfhClaimingAmountSummary.row(request.userAnswers, taxYear, businessId, user),
+              LiveAtBusinessPremisesSummary.row(request.userAnswers, taxYear, businessId, user),
+              BusinessPremisesAmountSummary.row(request.userAnswers, taxYear, businessId, user),
+              BusinessPremisesDisallowableAmountSummary.row(
+                request.userAnswers,
+                taxYear,
+                businessId,
+                user,
+                request.userAnswers.get(BusinessPremisesAmountPage, Some(businessId)).get),
+              LivingAtBusinessPremisesOnePersonSummary.row(request.userAnswers, taxYear, businessId, user),
+              LivingAtBusinessPremisesTwoPeopleSummary.row(request.userAnswers, taxYear, businessId, user),
+              LivingAtBusinessPremisesThreePlusPeopleSummary.row(request.userAnswers, taxYear, businessId, user),
+              WfbpFlatRateOrActualCostsSummary.row(request.userAnswers, taxYear, businessId, user, wfbpRate.getOrElse("")),
+              WfbpClaimingAmountSummary.row(request.userAnswers, taxYear, businessId, user)
+            )
+          )
+
+          Ok(
+            view(
+              WorkplaceRunningCostsCYAPage.toString,
+              taxYear,
+              request.userType,
+              summaryList,
+              routes.WorkplaceRunningCostsCYAController.onSubmit(taxYear, businessId)))
+
+        case _ => redirectJourneyRecovery()
+      }
+    }
+
+  def onSubmit(taxYear: TaxYear, businessId: BusinessId): Action[AnyContent] = (identify andThen getUserAnswers andThen requireData).async {
+    implicit request =>
+      val context = JourneyContextWithNino(taxYear, request.nino, businessId, request.mtditid, ExpensesWorkplaceRunningCosts)
+      val result  = service.submitAnswers[WorkplaceRunningCostsJourneyAnswers](context, request.userAnswers)
+
+      handleSubmitAnswersResult(context, result)
+  }
+
+  def getFlatRates(request: DataRequest[_], businessId: BusinessId): (Option[String], Option[String]) = {
+
+    val months25To50  = request.getValue(WorkingFromHomeHours25To50, businessId)
+    val months51To100 = request.getValue(WorkingFromHomeHours51To100, businessId)
+    val months101Plus = request.getValue(WorkingFromHomeHours101Plus, businessId)
+
+    val months1Person = request.getValue(LivingAtBusinessPremisesOnePerson, businessId)
+    val months2People = request.getValue(LivingAtBusinessPremisesTwoPeople, businessId)
+    val months3People = request.getValue(LivingAtBusinessPremisesThreePlusPeople, businessId)
+
+    (months1Person, months2People, months3People, months25To50, months51To100, months101Plus) match {
+      case (Some(months1Person), Some(months2People), Some(months3People), Some(months25To50), Some(months51To100), Some(months101Plus)) =>
+        val amount25To50  = months25To50 * 10
```

Comment:
I think it would be better to extract these calculations to some case class / object, so it can be independently unit tested.  It is not immediately obvious what these are.